rskj-core/src/main/java/org/ethereum/datasource/KeyValueDataSource.java

```diff
@@ -49,6 +50,8 @@ public interface KeyValueDataSource extends DataSource {
 
     Set<ByteArrayWrapper> keys();
 
+    DbIterator iterator();
```

Comment:
how about introducing a new interface like this?

```java
interface DataSourceKeyIterator extends Iterator<byte[]>, AutoCloseable { }
```

and returning it here instead of `DbIterator` ?

`DbIterator` doesn't provide any interface, so when being used you have to cast it to specific implementations

Reply:
Hmm, not sure how this would work for levelbd and rocksdb implementation. Their iterator implementation is different.

Reply:
Done